lua/wfp.lua

```diff
@@ -30,47 +30,100 @@ local atBus2Name = {
 
 
 
-local xml = require 'pl.xml'
 local WFPXml = require 'pl.class'()
-function WFPXml:_init(version, tLog)
+function WFPXml:_init(tLog)
     -- more information about pl.xml here: https://stevedonovan.github.io/Penlight/api/libraries/pl.xml.html
-    version = version or "1.3.0"
+    self.xml = require 'pl.xml'
+    self.nodeFlasherPack = nil
+    self.tTargets = {}
     self.tLog = tLog
-    self.nodeFlasherPack = xml.new("FlasherPackage")
+end
+
+function WFPXml:parse(strWfpXmlData)
+    self.nodeFlasherPack = self.xml.parse(strWfpXmlData, true, true)
+    -- self.tTarget = ???
+    self.tTargets = self.nodeFlasherPack:get_elements_with_name("Target")
+    -- local strXmlData = self.xml.tostring(self.nodeFlasherPack, "", "    ", nil, true)
+    -- print(strXmlData)
+end
+
+function WFPXml:get_target(strNetX)
+    for _, tTarget in ipairs(self.tTargets) do
+
+        if strNetX == tTarget["attr"]['netx'] then
+            return tTarget
+        end
+    end
+    return nil
+end
+
+function WFPXml:new(version)
+    version = version or "1.3.0"
+    self.nodeFlasherPack = self.xml.new("FlasherPackage")
     self.nodeFlasherPack:set_attrib("version", version)
     self.nodeFlasherPack:set_attrib("has_subdirs", "True")
 end
 
 function WFPXml:addTarget(strTargetName)
-	self.tTarget = xml.new("Target")
-	self.tTarget:set_attrib("netx", strTargetName)
-	self.nodeFlasherPack:add_child(self.tTarget)
+	local tTarget = self.xml.new("Target")
```

Comment:
Sidenote: using xml.new() will create a new XML document for each node. I think the correct way to create tags is xml.elem(). It's probably no problem though.